src/analysis/functions/mod.rs

```diff
@@ -0,0 +1,83 @@
+use r2api::structs::LSectionInfo;
+use petgraph::graph::NodeIndex;
+
+use frontend::radeco_containers::{RadecoFunction, RadecoModule};
+use middle::ir::MOpcode;
+use middle::ssa::ssa_traits::SSA;
+use middle::ssa::ssa_traits::SSAWalk;
+use middle::ssa::ssa_traits::NodeType;
+use middle::regfile::SubRegisterFile;
+
+use std::sync::Arc;
+use std::marker::PhantomData;
+
+pub struct FuncInfo {
+    arguments : Vec<NodeIndex>
+}
+
+pub trait FuncAnalysis: Transfer {}
+
+impl<T: Transfer> FuncAnalysis for T {}
+
+pub trait Transfer {
+    // Generalize to take anything (any information about the context/module)
+    // This function is only called the first time the analysis is executed. This function then
+    // returns `Self` that contains (partially-)analyzed information.
+    fn transfer(&mut RadecoFunction, &SubRegisterFile, &[LSectionInfo]) -> FuncInfo;
+}
+
+pub struct FunctionAnalyzer<T: FuncAnalysis> {
```

Comment:
Same as above, you shouldn't even be defining/implementing this.

The end result should look something that allows you to call: `reference_marking::InterProcAnalyzer<FunctionAnalyzer>`, where your `FunctionAnalyzer` would have implemented `Transfer` and `Propagate` and therefore should directly work with the `InterProcAnalyzer`.